src/buffer.rs

```diff
@@ -375,6 +376,54 @@ where
         let index = no_channels * (y * self.width + x) as usize;
         <P as Pixel>::from_slice(&self.data[index..index + no_channels])
     }
+
+    /// Return the raw sample buffer with its stride an dimension information.
+    ///
+    /// The returned buffer is guaranteed to be well formed in all cases. It is
+    /// layed out by colors, width then height, meaning `channel_stride <=
+    /// vertical_stride <= horizontal_stride`. All strides are in numbers of
+    /// elements but those are mostly `u8` in which case the strides are also
+    /// byte strides.
```

Comment:
Based on `get_pixels` I think you have this reversed. vertical_stride should be larger than horizontal_stride since changing x by one results in index changing by no_channels whereas changing y by 1 results in index changing by no_channels*width.

Reply:
You're right, I messed naming. `vertical_stride` is probably interpreted as going into the vertical/`y` direction. This then also implies a correction that is necessary in `FlatSamples::index`. I really should do tests soon.

Reply:
Speaking of which, should the `channel` parameter in `FlatSamples::index` be first? So that it the general order of indices consistent with `extents` and `ImageBuffer::dimensions()`